lib/util/password.rb

```diff
@@ -0,0 +1,13 @@
+module Util
+  # provides helper methods for validating password complexity
+  class Password
+    def self.validate_complexity(password)
+      # special characters according to https://www.owasp.org/index.php/Password_special_characters
+      #                    (2 uppercase letters)                 (  1 digit )                                              (12-128 characters)
+      #                                       (2 lowercase letters)          (          1 special character              )
+      unless password =~ /^(?=.*?[A-Z].*[A-Z])(?=.*?[a-z].*[a-z])(?=.*?[0-9])(?=.*[ !"#$%&'()*+,-.\/:;<=>?@\[\\\]^_`{|}~]).{12,128}$/
```

Comment:
can we move the regex to a const member of the class? This way it will not be calculated each time and this method will be more readable.

Reply:
If you do so, you can change this to a one-liner (raise error unless condition)

Reply:
moved to credentials and done